libs/geoservices/covid/src/lib/pages/maps/components/time-map/time-map.component.ts

```diff
@@ -0,0 +1,276 @@
+    requests.pipe(take(1)).subscribe(([sData, cData]) => {
+      this.stateData = sData[currentDateSelected];
+      this.countyData = cData[currentDateSelected];
+      this.selectedDate = currentDateSelected;
+      this.mortalButtonToggled ? this.drawDeathMap() : this.drawCasesMap();
+    });
+  }
+
+  public drawCasesMap() {
+    const stateExpression = ['match', ['get', 'STATE']];
+    const countyExpression = ['match', ['get', 'fips']];
+
+    this.stateData.forEach(function(row) {
+      const number = row['infection_rate'];
+      const color =
+        number > 1000
+          ? '#AE8080'
+          : number > 500
+          ? '#D18080'
+          : number > 100
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      stateExpression.push(row['STATE'], color);
+    });
+
+    this.countyData.forEach(function(row) {
+      const number = row['infection_rate'];
+      const color =
+        number > 1000
+          ? '#AE8080'
+          : number > 500
+          ? '#D18080'
+          : number > 100
+          ? '#FC8B8B'
+          : number > 10
+          ? '#FDC4C4'
+          : number > 1
+          ? '#FDE6E6'
+          : '#FFFFFF';
+      countyExpression.push(row['fips'], color);
+    });
+
+    stateExpression.push('rgba(255,255,255,1)');
+    countyExpression.push('rgba(255,255,255,1)');
+
+    this.mapService.loaded.pipe(take(1)).subscribe((map) => {
+      map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+      map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+    });
+  }
+
+  public drawDeathMap() {
+    const stateExpression = ['match', ['get', 'STATE']];
+    const countyExpression = ['match', ['get', 'fips']];
+
+    this.stateData.forEach(function(row) {
+      const number = row['death_rate'];
+      const color =
+        number > 100
+          ? '#54278F'
+          : number > 75
+          ? '#756BB1'
+          : number > 50
+          ? '#9E9AC8'
+          : number > 10
+          ? '#BCBDDC'
+          : number > 1
+          ? '#DADAEB'
+          : '#F2F0F7';
+      stateExpression.push(row['STATE'], color);
+    });
+
+    this.countyData.forEach(function(row) {
+      const number = row['death_rate'];
+      const color =
+        number > 100
+          ? '#54278F'
+          : number > 75
+          ? '#756BB1'
+          : number > 50
+          ? '#9E9AC8'
+          : number > 10
+          ? '#BCBDDC'
+          : number > 1
+          ? '#DADAEB'
+          : '#F2F0F7';
+      countyExpression.push(row['fips'], color);
+    });
+
+    stateExpression.push('rgba(255,255,255,1)');
+    countyExpression.push('rgba(255,255,255,1)');
+
+    this.mapService.loaded.pipe(take(1)).subscribe((map) => {
+      map.setPaintProperty('covid-state', 'fill-color', stateExpression);
+      map.setPaintProperty('covid-county', 'fill-color', countyExpression);
+    });
```

Comment:
Same here, consider eliminating the subscription altogether and do like you did in `switchLayers` method.

Reply:
resolved